util/updateXML/content/sources/Job/Jobs.py

```diff
@@ -265,35 +267,77 @@ def __get_update_modules_filepath_list(self, all_files_in_all_scripts_dirs):
+            module_version = self.__get_module_version(update_filepath)
+            if min_limit_version < tuple(map(int, module_version.split('.'))) < max_limit_version:
+                unsorted_update_modules_list.append(update_filepath)
+
+        if len(unsorted_update_modules_list) == 0:
+            print("Error : Patch files not found")
+            sys.exit(1)
+
+        sorted_update_modules_list = sorted(unsorted_update_modules_list, key=lambda update_filepath: os.path.basename(update_filepath))
+        return sorted_update_modules_list
+
+    def __extract_update_modules(self, all_files_in_all_scripts_dirs):
+        """
+        Get in a list all update scripts
+
+        Parameter:
+            all_files_in_all_scripts_dirs (list[str]): list of all files in selected directories without checking if
+                                                        they are update scripts or not
+
+        Returns:
+            all_update_filepaths_in_all_scripts_dirs (list[str]) : list of all update scripts
+        """
+        all_update_filepaths_in_all_scripts_dirs = list()
         invalid_update_files = list()
         main_update_filename = os.path.basename(self.__main_update_file_path)
         main_update_filename_without_extension = os.path.splitext(main_update_filename)[0]
-        for update_filepath in all_files_in_all_scripts_dirs:
-            update_file = os.path.basename(update_filepath)
-            if update_file.startswith(main_update_filename_without_extension) and \
-                    update_file.endswith(PYTHON_FILE_EXTENSION) and \
-                    update_file != main_update_filename:
-                update_module = os.path.splitext(update_file)[0]
+        for file_path in all_files_in_all_scripts_dirs:
+            file_name = os.path.basename(file_path)
+            if file_name.startswith(main_update_filename_without_extension) and \
+                    file_name.endswith(PYTHON_FILE_EXTENSION) and \
+                    file_name != main_update_filename:
+                update_module = os.path.splitext(file_name)[0]
                 module_version = update_module.split(main_update_filename_without_extension)[1]
-                if not re.match(r'^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$', module_version):
-                    invalid_update_files.append(update_file)
+                if not re.match(MODULE_VERSION_REGEX, module_version):
+                    invalid_update_files.append(file_name)
                     continue
-                min_limit_version = (self.__dynawo_origin[0], self.__dynawo_origin[1], self.__dynawo_origin[2]+1)
-                max_limit_version = (self.__dynawo_version[0], self.__dynawo_version[1], self.__dynawo_version[2]+1)
-                if min_limit_version < tuple(map(int, module_version.split('.'))) < max_limit_version:
-                    unsorted_update_modules_list.append(update_filepath)
+                all_update_filepaths_in_all_scripts_dirs.append(file_path)
```

Comment:
Can't we simply check that there is not duplicated basename in this list ?

Reply:
I thought using a function to check there is no duplicates would make the code more modular.
Moreover, with the check_duplicate_basenames() we can print all the duplicates. If we just check there is no duplicates when we add a file path, we'll just check one basename at a time.